matgen/generate_matrix_he.cc

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2017-2022, University of Tennessee. All rights reserved.
+// SPDX-License-Identifier: BSD-3-Clause
+// This program is free software: you can redistribute it and/or modify it under
+// the terms of the BSD 3-Clause license. See the accompanying LICENSE file.
+
+#include "slate/slate.hh"
+#include "test.hh"
+#include "print_matrix.hh"
+
+#include <exception>
+#include <string>
+#include <vector>
+#include <limits>
+#include <complex>
+#include <chrono>
+
+#include <cmath>
+#include <cstdio>
+#include <cstdlib>
+#include <utility>
+
+#include "matrix_params.hh"
+#include "generate_matrix.hh"
+#include "random.hh"
+
+// -----------------------------------------------------------------------------
+/// Generates an m-by-n Hermitian-storage test matrix.
+/// Handles Hermitian matrices.
+/// Diagonal elements of a Hermitian matrix must be real;
+/// their imaginary part must be 0.
+/// @see generate_matrix
+/// @ingroup generate_matrix
+///
+template <typename scalar_t>
+void generate_matrix(
```

Comment:
Since this is so short, we can probably just include it in `generate_matrix_tz.cc`.

You can also consider putting the generators for `geev`, `geevx`, `heev`, and `svd` into a single header.  They're all closely related, and half of them are just un-implemented stubs.

Reply:
Actually, I feel that keeping related code together is a bigger benefit, which helps us keep similar functions in sync.  Fewer files also makes it easier to remember which file a particular function belongs to, and to use autocomplete on file names.  It's not a big deal either way, just an organizational preference.

I'm definitely on board with combining `geev` and `geevx` into a single file since they both build a matrix from a spectral decomposition.

Reply:
These are good points. I can working on condensing the files more. I will combine files`generate_matrix_he.cc` and `generate_matrix_tz.cc` which I will name for now`generate_matrix_he_and_tz.cc`. The other option I think is I could combine `geev`, `geevx`, and `heev` into a single file called `generate_type_evalue_decomp.hh` since all three generate a matrix using eigenvalue decomposition.

Reply:
I would keep `generate_matrix_geev.hh` (geev, geevx), `generate_matrix_heev.hh`, and `generate_matrix_svd.hh`. Some of those are just stubs (geev, geevx), but will eventually be full fledged functions.

Reply:
Currently I have done the following based off this thread: 

- Combined `geev` and `geevx` into a single file titled `generate_type_geev.hh`
- Kept `generate_type_heev.hh` and `generate_type_svd.hh`as individual files
- Combined `generate_matrix_he.cc` and `generate_matrix_tz.cc` into `generate_matrix_he_and_tz.cc`

If not mentioned here such as `jordan`, `rand`, etc. then it has stayed as an individual file just like `heev` and `svd`.